### PR TITLE
Use Dask for qsub and parallell jobs

### PR DESCRIPTION
**Pull request by vinferrer**
Closes # .

Changes proposed in this pull request:

- Now `dask` is  used to launch `compute_stability.py` and the even_detection on the surrogates this way we avoid `bash` scripts and make easier to implement `connPFM` in other clusters
-

### CODE REVIEW

File: connPFM/deconvolution/roiPFM.py [hunk 1]
```diff
@@ -59,8 +57,8 @@ def roiPFM(
     hrf = hrf_matrix.hrf_norm
     LGR.info("HRF generated.")
     LGR.info("Running stability selection on original data...")
-    if which("singularity") is not None:
-        fetch_file("n7tzh", os.path.dirname(os.path.realpath(__file__)), "connpfm_slim.simg")
+    if jobs != 0:
+        _, cluster = dask_scheduler(jobs)
```
Comment (eurunuela) [inline connPFM/deconvolution/roiPFM.py:61]: Why are you not using `try:` here?

File: connPFM/deconvolution/compute_slars.py [hunk 2]
```diff
@@ -158,7 +69,7 @@ def main(argv):
             "AUC of voxel {}/{} calculated and stored...".format(str(vox_idx + 1), str(nvoxels))
         )
 
-    filename = args.tempdir[0] + "/auc_" + str(n_job) + ".npy"
+    filename = tempdir + "/auc_" + str(n_job) + ".npy"
```
[the remark this answers is not in the document]
Comment (vinferrer) [reply connPFM/deconvolution/compute_slars.py:72]: Oh sure, I just modify what I needed

**Comment by vinferrer** [comment]
Thinking that we may be using different clusters I changed `dask_scheduler` function to detect if there is a `jobqueue.yaml` file and get from there which scheduler is used

File: connPFM/utils/io.py [hunk 3]
```diff
@@ -63,12 +65,44 @@ def save_img(data, output, masker, history_str=None):
...
 def dask_scheduler(jobs):
-    config.set(distributed__comm__timeouts__tcp="90s")
-    config.set(distributed__comm__timeouts__connect="90s")
-    config.set(scheduler="single-threaded")
-    config.set({"distributed.scheduler.allowed-failures": 50})
-    config.set(admin__tick__limit="3h")
-    cluster = SGECluster(memory="20Gb")  # assumees dask configuration file
-    client = Client(cluster)
-    cluster.scale(jobs)
+    # look if default ~ .config/dask/jobqueue.yaml exitsts
```
Comment (eurunuela) [inline connPFM/utils/io.py:68]:
```suggestion
    # look if default ~ .config/dask/jobqueue.yaml exists
```

File: connPFM/utils/io.py [hunk 4]
```diff
@@ -63,12 +65,44 @@ def save_img(data, output, masker, history_str=None):
 
 
 def dask_scheduler(jobs):
```
Comment (eurunuela) [inline connPFM/utils/io.py:67]: We have to add the docstring here

File: connPFM/utils/io.py [hunk 5]
```diff
@@ -63,12 +65,44 @@ def save_img(data, output, masker, history_str=None):
...
+        )
+        cluster = None
+    else:
+        config.set(distributed__comm__timeouts__tcp="90s")
+        config.set(distributed__comm__timeouts__connect="90s")
+        config.set(scheduler="single-threaded")
+        config.set({"distributed.scheduler.allowed-failures": 50})
+        config.set(admin__tick__limit="3h")
+        if "sge" in data["jobqueue"]:
+            cluster = SGECluster(memory="20Gb")
```
Comment (eurunuela) [inline connPFM/utils/io.py:87]: Shouldn't the memory be a parameter users input?

**Review by eurunuela — CHANGES_REQUESTED** [review]
I have a couple of comments I'd like resolved before merging.

File: connPFM/utils/io.py (hunk 5 above)
Comment (vinferrer) [reply connPFM/utils/io.py:87]: In any case i would leave it empty and the user can modify this in the jobqueue.yaml file. I can actually add an example file

Comment (eurunuela) [reply connPFM/utils/io.py]: I think having an example would be great.

**Review by eurunuela — APPROVED** [review]
LGTM! Do you want to add the example yaml file in this PR or would you rather open a new one?

**Comment by vinferrer** [comment]
Thats already in the utils folder

**Pull request merged**